anvill/src/Lifters/ValueLifter.cpp

```diff
@@ -50,12 +51,74 @@ llvm::APInt ValueLifter::ConsumeBytesAsInt(std::string_view &data,
   }
 }
 
+llvm::Constant *ValueLifter::GetPointer(uint64_t ea, llvm::Type *data_type,
+                                        EntityLifterImpl &ent_lifter) const {
+
+  llvm::GlobalValue *found_entity_at = nullptr;
+  auto &func_lifter = ent_lifter.function_lifter;
+
+  ent_lifter.ForEachEntityAtAddress(
+      ea, [&](llvm::GlobalValue *gv) { found_entity_at = gv; });
+
+  if (found_entity_at) {
+    return llvm::ConstantExpr::getBitCast(found_entity_at, data_type);
+  }
+
+  const auto pointer_type = llvm::dyn_cast<llvm::PointerType>(data_type);
+  const auto elm_type = pointer_type->getElementType();
+
+  if (elm_type->getTypeID() == llvm::Type::FunctionTyID) {
+
+    // Get the function at the given address; if it is missing from the spec
+    // fallback to returning the address
+    auto maybe_decl = ent_lifter.type_provider->TryGetFunctionType(ea);
+    if (maybe_decl) {
+      auto func = func_lifter.DeclareFunction(*maybe_decl);
+      auto func_in_context =
+          func_lifter.AddFunctionToContext(func, ea, ent_lifter);
+
+      // Getting wrong function pointer type here. Check??
+      return llvm::ConstantExpr::getBitCast(func_in_context, pointer_type);
+    } else {
+      LOG(ERROR) << "Failed to get function at address " << std::hex << ea
+                 << std::hex << " missing from the spec file!";
+    }
+
+  } else {
+    auto maybe_decl = ent_lifter.type_provider->TryGetVariableType(ea, dl);
+    if (maybe_decl) {
+
+      // if the variable start address matches with ea
+      if (maybe_decl->address == ea) {
+        auto var =
```

Comment:
I think you always want to do this `auto var =`, then you want to build a pointer to the offset of this thing, and also add it into the lifter context as an entity at address `ea`.